source/slang/slang-check-decl.cpp

```diff
@@ -1767,11 +1768,22 @@ namespace Slang
     {
         if (auto initExpr = varDecl->initExpr)
         {
-            // If the variable has an explicit initial-value expression,
-            // then we simply need to check that expression and coerce
-            // it to the type of the variable.
-            //
-            initExpr = CheckTerm(initExpr);
+            if (std::getenv("RUN") &&
+                varDecl->hasModifier<HLSLStaticModifier>() &&
+                this->m_outerDecl == nullptr)
```

Comment:
why m_outerDecl==nullptr is needed?

Reply:
For now let's check for both `static` and `const` modifier.

Reply:
The setup for m_outerDecl can be done independently to whether or not it is static const.
Consider renaming it to m_outerVarDecl and always set the state.

Reply:
> why m_outerDecl==nullptr is needed?

No need, will remove.

Reply:
> For now let's check for both `static` and `const` modifier.

I didn't find static modifer in slang-ast-modifier.h, only see ConstModifier. Should we define one?

Reply:
Never mind, I thought we have a modifier for StaticConstModifer.
I changed to check HLSLStaticModifier and ConstModifier.: https://github.com/shader-slang/slang/pull/3635/commits/9897ff42b1d9536006cf68b0097f4044210d01a4